sklearnex/spmd/basic_statistics/tests/test_incremental_basic_statistics_spmd.py

```diff
@@ -0,0 +1,291 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+from onedal.basic_statistics.tests.test_basic_statistics import options_and_tests
+from onedal.tests.utils._dataframes_support import (
+    _convert_to_dataframe,
+    get_dataframes_and_queues,
+)
+from sklearnex.tests._utils_spmd import (
+    _generate_statistic_data,
+    _generate_weights,
+    _get_local_tensor,
+    _mpi_libs_and_gpu_available,
+)
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("weighted", [True, False])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.mpi
+def test_incremental_basic_statistics_fit_spmd_gold(dataframe, queue, weighted, dtype):
+    # Import spmd and batch algo
+    from sklearnex.basic_statistics import IncrementalBasicStatistics
+    from sklearnex.spmd.basic_statistics import (
+        IncrementalBasicStatistics as IncrementalBasicStatistics_SPMD,
+    )
+
+    # Create gold data and process into dpt
+    data = np.array(
+        [
+            [0.0, 0.0, 0.0],
+            [0.0, 1.0, 2.0],
+            [0.0, 2.0, 4.0],
+            [0.0, 3.0, 8.0],
+            [0.0, 4.0, 16.0],
+            [0.0, 5.0, 32.0],
+            [0.0, 6.0, 64.0],
+            [0.0, 7.0, 128.0],
+        ],
+        dtype=dtype,
+    )
+    dpt_data = _convert_to_dataframe(data, sycl_queue=queue, target_df=dataframe)
+
+    local_dpt_data = _convert_to_dataframe(
+        _get_local_tensor(data), sycl_queue=queue, target_df=dataframe
```

Comment:
Please be explicit with dtype in _convert_to_dataframe as per the other comment

Reply:
`_convert_to_dataframe` this will not work for array api for example, array api case should be updated in `_convert_to_dataframe`.
Even explicitly provided dtype doesn't guarantee that this will be the same dtpye. I think, asserts after `_convert_to_dataframe` more preferable.

Reply:
i still can't get why should it be only here and not in tons of other occurrences of this call in our repo. in my understanding it will become completely unnecessary after we test _convert_to_dataframe separately which is already planned. moreover, dtype check of the  _convert_to_dataframe result is not straightforward as far as i understand because the returning value may have different types depending on arguments of convert_to_dataframe. so it would lead to perceptible test size increase which i don't like and don't think it should be here